tests/run-make/split-debuginfo/rmake.rs

```diff
@@ -0,0 +1,428 @@
+            split_dwarf: SplitDwarf::Single,
+            has_o: true,
+            has_dwo: false,
+            dwp: None,
+            output: Some(bin_name("foo")),
+            remap_scope: Some(Some(Scope::Debug)),
+        });
+        run_test(DebugInfoTest {
+            input: "foo.rs",
+            crate_type: None,
+            debug_info: DebugInfo::Unpacked,
+            split_dwarf: SplitDwarf::Single,
+            has_o: true,
+            has_dwo: false,
+            dwp: None,
+            output: Some(bin_name("foo")),
+            remap_scope: Some(Some(Scope::Macro)),
+        });
+
+        run_test(DebugInfoTest {
+            input: "bar.rs",
+            crate_type: Some(CrateType::Lib),
+            debug_info: DebugInfo::Unpacked,
+            split_dwarf: SplitDwarf::Split,
+            has_o: false,
+            has_dwo: true,
+            dwp: None,
+            output: None,
+            remap_scope: None,
+        });
+        run_test(DebugInfoTest {
+            input: "main.rs",
+            crate_type: None,
+            debug_info: DebugInfo::Unpacked,
+            split_dwarf: SplitDwarf::Split,
+            has_o: false,
+            has_dwo: true,
+            dwp: None,
+            output: Some(bin_name("main")),
+            remap_scope: None,
+        });
+
+        run_test(DebugInfoTest {
+            input: "bar.rs",
+            crate_type: Some(CrateType::Lib),
+            debug_info: DebugInfo::Unpacked,
+            split_dwarf: SplitDwarf::Single,
+            has_o: true,
+            has_dwo: false,
+            dwp: None,
+            output: None,
+            remap_scope: None,
+        });
+        run_test(DebugInfoTest {
+            input: "main.rs",
+            crate_type: None,
+            debug_info: DebugInfo::Unpacked,
+            split_dwarf: SplitDwarf::Single,
+            has_o: true,
+            has_dwo: false,
+            dwp: None,
+            output: Some(bin_name("main")),
+            remap_scope: None,
+        });
+    }
+}
+
+#[track_caller]
+fn run_test(params: DebugInfoTest) {
+    let mut compiler = unstable_rustc();
+    compiler.input(params.input).arg("-Cdebuginfo=2");
+    match params.debug_info {
+        DebugInfo::Packed => compiler.arg("-Csplit-debuginfo=packed"),
+        DebugInfo::Unpacked => compiler.arg("-Csplit-debuginfo=unpacked"),
+    };
+    match params.split_dwarf {
+        SplitDwarf::Split => compiler.arg("-Zsplit-dwarf-kind=split"),
+        SplitDwarf::Single => compiler.arg("-Zsplit-dwarf-kind=single"),
+    };
+    if let Some(crate_type) = params.crate_type {
+        match crate_type {
+            CrateType::Rlib => compiler.crate_type("rlib"),
+            CrateType::Lib => compiler.crate_type("lib"),
+        };
+    };
+    if params.input == "lto.rs" {
+        compiler.arg("-Clinker-plugin-lto");
+    }
+    if params.input == "main.rs" {
+        compiler.extern_("bar", rust_lib_name("bar"));
+    }
```

Comment:
Suggestion: matching conditional logic based on input file name seems a bit weird, could we rework this to use an enum so it's more obvious at the callsite?